src/uu/wc/src/wc.rs

```diff
@@ -648,22 +808,27 @@ macro_rules! record_error {
     }
 }
 
-fn wc(inputs: &[Input], settings: &Settings) -> UResult<()> {
+fn wc(inputs: &Inputs, settings: &Settings) -> UResult<()> {
     let mut total_word_count = WordCount::default();
+    let mut num_inputs: usize = 0;
 
     let (number_width, are_stats_visible) = match settings.total_when {
         TotalWhen::Only => (1, false),
         _ => (compute_number_width(inputs, settings), true),
     };
-    let is_total_row_visible = settings.total_when.is_total_row_visible(inputs.len());
 
-    for (idx, input) in inputs.iter().enumerate() {
-        if matches!(input, Input::Path(p) if p.as_os_str().is_empty()) {
-            let err = WcError::zero_len(settings.files0_from_path.as_deref().map(|s| (s, idx)));
-            record_error!("{err}");
-            continue;
-        }
-        let word_count = match word_count_from_input(input, settings) {
+    for maybe_input in inputs.try_iter(settings)? {
+        num_inputs = num_inputs.saturating_add(1);
```

Comment:
This seems a bit overly defensive. If `usize` is gonna overflow, we have other more important problems :)